### PR TITLE
Firebase init

### LINKED ISSUE #7: Connect Firebase

### PR DESCRIPTION
**Pull request by ianmah**
Closes #7

### CODE REVIEW

File: src/utility/firebase.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import firebase from 'firebase/app'
+import 'firebase/firestore'
+
+if (!firebase.apps.length) {
```
Comment (jackyzha0) [inline src/utility/firebase.js:4]: What is this conditional checking? To see if there's a firebase app?

File: src/App.js [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
-const ExampleText = styled.div`
-  color: ${p => p.theme.colors.text}
-`
 
 function App() {
+  const [hackathons, setHackathons] = useState()
+
+  // example getting all hackathons data
+  useEffect(() => {
+    db.collection('Hackathons').get().then(querySnapshot => {
```
Comment (jackyzha0) [inline src/App.js:16]: Would it make sense to make a collection for each hackathon so we don't grab more info than we need?

File: src/App.js [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
+      const paths = querySnapshot.docs.map(doc => {
```
Comment (jackyzha0) [inline src/App.js:17]: Why is this var called paths

File: src/utility/firebase.js (hunk 1 above)
Comment (ianmah) [reply src/utility/firebase.js:4]: Yeah, this file will be imported many times and we only want a single app

File: src/App.js (hunk 2 above)
Comment (ianmah) [reply src/App.js:16]: This is just an example call, we can defs just grab one hackathon's data.

File: src/utility/firebase.js (hunk 1 above)
Comment (jackyzha0) [reply src/utility/firebase.js:4]: Maybe a good place to use a singleton?

Comment (ianmah) [reply src/utility/firebase.js:4]: this is the singleton

Comment (jackyzha0) [reply src/utility/firebase.js:4]: oh lmao all the singletons i've seen before had a slightly different structure but this works

**Pull request merged**